modules/c++/math/include/math/Utilities.h

```diff
@@ -51,6 +52,25 @@ inline double square(double val)
     return val * val;
 }
 
+/*!
+ * Return true if argument is NaN
+ *
+ * \param value Argument to be checked for NaN
+ * \return true if value is NaN
+ */
+template <typename T> bool isNaN(T value)
+{
+#ifdef HAVE_ISNAN
+    #include <math.h>
```

Comment:
Put this up with the other includes.  I don't remember what happens if you include a header inside a namespace other than to say that there are times where it really really really doesn't do what you want, so you always want to put them up top.  For simplicity I wouldn't bother to put it inside a new ifdef up top; always including it is fine.